### ISSUE TITLE
Incorrect base_64 messages from real world

### ISSUE THREAD

**Issue by AlexBekhtin** [barbushin/php-imap#88]
I have a problem to read mail body (example: HP OpenView autogenerated message).
in Mailbox.php used imap_base64
```php
		elseif($partStructure->encoding == 3) {
			$data = imap_base64($data);
		}
```
but if data contains incorrect characters result is 'false'.  MS Outlook and other mail programs can view this message. if I use base64_decode function, message decoded correctly.
Example:
```php
print ("imap_base64 correct:\n");
print (imap_base64('dGVzdCBtYWlsIHRleHQ=') . "\n");
print ("imap_base64 incorrect:\n");
print (imap_base64('dGVzdCBtYWlsIHRleHQ  .  ') . "\n");

print ("base64_decode correct:\n");
print (base64_decode('dGVzdCBtYWlsIHRleHQ=') . "\n");
print ("base64_decode incorrect:\n");
print (base64_decode('dGVzdCBtYWlsIHRleHQ  .  ') . "\n");
```

**Comment by barbushin** [comment]
Why there is ` . ` in the end of data string? You can try to fix it using ` $data = imap_base64(rtrim($data, ' .'));`.

**Comment by AlexBekhtin** [comment]
Mail data incorrect. It's true. I know it. I have more and more real mail and php_imap get data with incorrect characters. Real example:
```php
print (imap_base64(
'wuD4IO7y4uXyIO/uIO7h8ODh7vLq5SDu4fDg+eXt6P8gNzU0MjMyIO/w6O3/8i4g0e/g8ejh7iDn
4CDx7vLw8+Tt6Pfl8fLi7iEKCi0tLQrd8u4g8e7u4fnl7ejlIOH76+4g8ePl7eXw6PDu4uDt7iDg
4vLu7ODy6Pfl8eroLiDP7ubg6/Pp8fLgLCDt5SDu8uLl9+Dp8uUg7eAg7eXj7i4K


.


'
));
```
This string is wrong. All e-mail programs read this message correctly. But it generated by HP OpenView and I have 2 ways: 1) Path manualy to use base64_decode function (or try use trim) 2) write this issue. I have no big statistic.

**Comment by georaldc** [comment]
Looking at the php docs, imap_base64() seems to be similar to the base64_decode() function with the strict argument set to true. Meaning you get false returned if your encoded string contains anything outside the base64 alphabet.

Not the answer you are looking for maybe, but imap_base64() might be working as intended in this case.

**Comment by barbushin** [comment]
Released in 2.0.3
